### PR TITLE
Round cart condition values

### PR DESCRIPTION
**Pull request by ryanmitchell**

### CODE REVIEW

File: src/Cart/Helpers/CartConditionHelper.php [hunk 1]
```diff
@@ -50,7 +50,8 @@ protected function processActionValue($action, $total)
             $value = (float)$this->cleanValue($actionValue);
         }
 
-        $this->calculatedValue += $value;
+        $precision = app('currency')->getDefault() ? app('currency')->getDefault()->decimal_position : 2;
+        $this->calculatedValue += round($value, $precision);
```
Comment (sampoyigi) [inline src/Cart/Helpers/CartConditionHelper.php:54]: Let's pass the currency precision to the Cart library/CartCondition as config or property.

Should we also implement precision when calculating any money values within the Cart library.. For example, when calculating the cart item total?

**Review by sampoyigi — CHANGES_REQUESTED** [review]

**Comment by ryanmitchell** [comment]
I dont think it does any harm to round it whenever we can. Certainly helps from a tax calculation point of view. If you can make a rough list of where and I’ll work through them.

In terms of passing the precision - is that not what app() helps you get around? You know laravel better than me :)

**Comment by sampoyigi** [comment]
I'd say we tackle the rounding issue across the platform another time. 

Yup let's move this to the tax cart condition class and round the taxRate value.

No worries with passing the precision, we can use the currency service within the tax class.

**Comment by ryanmitchell** [comment]
No worries - so back this out and just make it specific to the tax condition?

**Comment by sampoyigi** [comment]
> No worries - so back this out and just make it specific to the tax condition?

Yup

**Comment by ryanmitchell** [comment]
Opened a PR In cart

**Pull request merged**